controllers/db/reports/publication.report.search.controller.ts

```diff
@@ -0,0 +1,1453 @@
+                      model: models.PersonPersonType,
+                      as: "PersonPersonTypes",
+                      required: true,
+                      // separate : true,
+                      on: {
+                        col1: Sequelize.where(Sequelize.col('AnalysisSummaryAuthor.personIdentifier'), "=", Sequelize.col('PersonPersonTypes.personIdentifier')),
+                       // col2: Sequelize.where(Sequelize.col('Person.personIdentifier'), "=", Sequelize.col('PersonPersonTypes.personIdentifier'))
+                        },
+                      // where : joinWherePersonTypes,
+                      attributes: [`personIdentifier`],
+                    },
+                  ],
+                  where: joinWhereAuthorsFilters,
+                  subQuery: false,
+                  // limit: apiBody.limit,
+                  // offset: apiBody.offset,
+                  benchmark: true
+                });
+              }
+            }
+            else  { // When author filters present and no PersonType
+              results = await models.AnalysisSummaryAuthor.findAndCountAll({ // review commnet:Change this to findAll
+                  attributes: ["personIdentifier", "pmid"],
+                  distinct:true,
+                  col:'pmid',
+                  include: [
+                    {
+                      model: models.Person,
+                      as: "Person",
+                      required: true,
+                      on: {
+                        col: Sequelize.where(Sequelize.col('AnalysisSummaryAuthor.personIdentifier'), "=", Sequelize.col('Person.personIdentifier'))
+                        },
+                      // where : joinWhereAuthorsFilters,
+                      attributes: [],
+                    }
+                  ],
+                  where: joinWhereAuthorsFilters,
+                  subQuery: false,
+                  // limit: apiBody.limit,
+                  // offset: apiBody.offset,
+                  benchmark: true
+                });
+              }
+      }else if(!isAuthorFilter && isArticleFilter){   // Filter Only with Articles and not Authors
+        results = await models.AnalysisSummaryArticle.findAndCountAll({
+          include:[{
+            model: models.AnalysisSummaryAuthor,
+            as: "AnalysisSummaryAuthor",
+            required: true,
+            on: {
+              col: Sequelize.where(
+                Sequelize.col("AnalysisSummaryAuthor.pmid"),
+                "=",
+                Sequelize.col("AnalysisSummaryArticle.pmid")
+              ),
+            },
+            attributes: ["personIdentifier","pmid"],
+          }],
+          where: whereForOnlyArticles,
+          subQuery: false,
+          // limit: apiBody.limit,
+          // offset: apiBody.offset,
+          order: sort,
+         // group: ["AnalysisSummaryArticle.pmid"], // This grouping is not required as we need only count of records statisfies criteria
```

Comment:
Same as above lets remove this commented out code and any commented out code after this